src/attribution/onLCP.ts

```diff
@@ -54,9 +54,13 @@ const attributeLCP = (metric: LCPMetric): LCPMetricWithAttribution => {
               activationStart
           : 0,
       );
-      const lcpResponseEnd = Math.max(
-        lcpRequestStart,
-        lcpResourceEntry ? lcpResourceEntry.responseEnd - activationStart : 0,
+      const lcpResponseEnd = Math.min(
+        // Cap at LCP time (videos continue downloading after LCP for example)
+        lcpEntry.startTime - activationStart,
+        Math.max(
+          lcpRequestStart,
+          lcpResourceEntry ? lcpResourceEntry.responseEnd - activationStart : 0,
+        ),
```

Comment:
Besides handling the video case, clamping on LCP does feel like a more correct constraint if trying to fix up a broken ordering of these numbers (probably a different argument to be made to not fix illogical timings and just report them as an error instead, but conversation for another day :)

Note that it makes the `Math.max` for `lcpRenderTime` below redundant (it'll always be `lcpEntry.startTime - activationStart`), so that should be removed, but treating the reported LCP as a fixed point rather than having the subpart breakdown not add up to LCP seems like a better thing to do.

The constraints still aren't quite right, though. e.g. an LCP before `lcpRequestStart` will win out now, placing `lcpResponseEnd` and `lcpRenderTime` before `lcpRequestStart`, leading to a negative load duration.

It might be helpful to pull out some intermediate variables to make this easier to reason about and to read? e.g. define `rawLcpRequestStart` and `rawLcpResponseEnd` (or named however) directly, and then apply the max/mins to ensure they're ordered correctly.

Reply:
> Note that it makes the `Math.max` for `lcpRenderTime` below redundant (it'll always be `lcpEntry.startTime - activationStart`), so that should be removed

Good spot! Done.

> but treating the reported LCP as a fixed point rather than having the subpart breakdown not add up to LCP seems like a better thing to do.

I'm not getting what you mean?

> The constraints still aren't quite right, though. e.g. an LCP before lcpRequestStart will win out now, placing lcpResponseEnd and lcpRenderTime before lcpRequestStart, leading to a negative load duration.

I can't see how that would happen though? Or are you thinking about a video when the poster image is loaded before the video? But in that case the LCP resource should be the poster image.

> It might be helpful to pull out some intermediate variables to make this easier to reason about and to read? e.g. define rawLcpRequestStart and rawLcpResponseEnd (or named however) directly, and then apply the max/mins to ensure they're ordered correctly.

Yeah could be made clearer. Will park this for now while we thrash out the actual implementation.

Reply:
> > but treating the reported LCP as a fixed point rather than having the subpart breakdown not add up to LCP seems like a better thing to do.
> 
> I'm not getting what you mean?

before it could bump up `lcpRenderTime` to a later point, even if after LCP, so it seems better to keep `lcpResponseEnd` at `lcpRenderTime` or below, as your change does.

> > The constraints still aren't quite right, though. e.g. an LCP before lcpRequestStart will win out now, placing lcpResponseEnd and lcpRenderTime before lcpRequestStart, leading to a negative load duration.
> 
> I can't see how that would happen though? Or are you thinking about a video when the poster image is loaded before the video? But in that case the LCP resource should be the poster image.

No idea how it could happen, I'm just saying that's what the code says it can do :) Presumably we're clamping to each value because we think it could happen.

Reply:
> Presumably we're clamping to each value because we think it could happen.

Yeah but we have reasons why we think they could happen.

> It might be helpful to pull out some intermediate variables to make this easier to reason about and to read? e.g. define rawLcpRequestStart and rawLcpResponseEnd (or named however) directly, and then apply the max/mins to ensure they're ordered correctly.

Let this for now as `lcpResponseEnd` is similar to `lcpRequestStart` above.